### PR TITLE
Make Server Logger Implementation Generic

### PR DESCRIPTION
**Pull request by ChristopherDavenport**

### CODE REVIEW

**Comment by ChristopherDavenport** [comment]
Note: This buffers on RequestLogger despite never logging anything for it on anything on it for this Route. I don't have a great way to get around this at present.

File: server/src/main/scala/org/http4s/server/middleware/Logger.scala [hunk 1]
```diff
@@ -15,21 +17,37 @@ import org.log4s.getLogger
...
       Sync[F].delay(logger.info(s))
     }
-    ResponseLogger(logHeaders, logBody, redactHeadersWhen, log.pure[Option])(
-      RequestLogger(logHeaders, logBody, redactHeadersWhen, log.pure[Option])(http)
+    ResponseLogger(logHeaders, logBody, fk, redactHeadersWhen, log.pure[Option])(
+      RequestLogger(logHeaders, logBody, fk, redactHeadersWhen, log.pure[Option])(http)
     )
   }
 
+  def httpApp[F[_]: Concurrent](
```
Comment (rossabaker) [inline server/src/main/scala/org/http4s/server/middleware/Logger.scala]: We haven't specialized our others that need an `~>` like this.  I think we should do all or none.  Not strongly opinionated which.

**Comment by ChristopherDavenport** [comment]
My biggest point of contention has been that there exists ping and other informational routes that we would prefer to explicitly NOT log as they are noisy.

I'm not really picky as far as our implementation of applying the `~>` or not. Just seemed easier to me to have default cases, as otherwise to use a basic logger people would need to know about `FunctionK`. Which is a high bar to set for logging http interactions. However, that also makes our api more verbose. I think perhaps we should add more of these moving forward.

**Comment by ChristopherDavenport** [comment]
Plus it gives us some low hanging fruit issues which are always nice.

**Pull request merged**